### PR TITLE
Node 6: add Firebase snippets

### PR DESCRIPTION
**Pull request by ace-n**
**Do not merge** until we figure out the following:
1) Are these needed within our docs? **Yes**
2) Do they work with Node 8? **No**, they depend on a callback. We'll have to add Node 8 samples separately.
3) Does the analytics trigger _actually work_? **No**, so we'll have to revisit it later.

### CODE REVIEW

**Comment by ace-n** [comment]
@mogar1980 FYI

File: functions/firebase/index.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  console.log(JSON.stringify(event));
+
+  // Don't forget to call the callback.
+  callback();
+};
+// [END functions_firebase_firestore]
+
+// [START functions_firebase_auth]
+/**
+ * Triggered by a change to a Firebase Auth user object.
```
Comment (samtstern) [inline functions/firebase/index.js]: I think we only have triggers for user creation and deletion right now, not on any change to the user object.

File: functions/firebase/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    console.error(err);
+  }
+  // Don't forget to call the callback.
+  callback();
+};
+// [END functions_firebase_auth]
+
+// [START functions_firebase_analytics]
+/**
+ * Triggered by a Firebase Mobile Analytics log event.
```
Comment (samtstern) [inline functions/firebase/index.js]: We call it "Google Analytics for Firebase"

**Comment by samtstern** [comment]
Two small comments but LGTM (assuming they actually work, I did not take the time to deploy them)

**Pull request merged**